rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -23,10 +25,21 @@ public JsonError resolveError(Throwable t, Method method, List<JsonNode> argumen
             error =  new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
         } else if (t instanceof InvalidFormatException) {
             error = new JsonError(-32603, "Internal server error, probably due to invalid parameter type", null);
+        } else if (t instanceof UnrecognizedPropertyException) {
+            String message = getExceptionMessage((UnrecognizedPropertyException) t);
+            logger.error("JsonRPC error {} for method {} with arguments {}", message, method, arguments);
+            error = new JsonError(-32603, message, null);
         } else {
             logger.error("JsonRPC error when for method {} with arguments {}", method, arguments, t);
             error = new JsonError(-32603, "Internal server error", null);
         }
         return error;
     }
+
+    private String getExceptionMessage(UnrecognizedPropertyException ex) {
+        if (ex.getMessage() == null) return "Invalid parameters";
+        return Arrays.stream(ex.getMessage().split("\\n at "))
```

Comment:
as an alternative we may use `getPropertyName()` of `UnrecognizedPropertyException` to compose the message (instead of parsing)

Reply:
Yes, I don't really like the exception message parsing, the problem is that the idea is to show a nice descriptive message (for example, including the allowed fields). I will see what can be done regarding that and update it accordingly.

Reply:
This idea of @Vovchyk was a nice one, so I've updated the message, now it is built using more the exception information: not only the property name but also the "known fields".

So, instead of showing this message:

`Unrecognized field "form" (class org.ethereum.rpc.Web3$CallArguments), not marked as ignorable (8 known properties: "gasPrice", "value", "gas", "from", "to", "chainId", "nonce", "data"])`

We show this:

`Unrecognized field "form" (8 known properties: ["gasPrice", "value", "gas", "from", "to", "chainId", "nonce", "data"])`

It is a subtle difference I think, but we hide internal details about the implementation and also fix an issue of the exception where the opening bracket ("[") was missing from the message.

Also, **some important thing**: now we are returning `-32602` (Invalid Parameters) instead of `-32603` (Internal Error).

@pmprete how do you see it know? Do you like the new message? And how about changing the status code? Thanks in advance!